### DISCUSSION TITLE
Why is `map()` used instead of `forEach()` in the `src/index.ts`?

### DISCUSSION THREAD

**Discussion by nikolausliu** [developit/mitt#190] [Q&A] [answered]
see this line: [src/index.ts L108](https://github.com/developit/mitt/blob/6b41670516ed8e8b738612f60491995470aa63b3/src/index.ts#L108)

@developit 
sorry for bothering. I'm a little confused，colud u help?

The above code is only used to traverse and does not care about the return value，but using `map()` instead of `forEach()`

I guess it's to reduce code size. Is there any deeper reason?

**Comment by developit** [comment, marked as answer]
This is done to reduce file size. "map" is shorter than "forEach".

**Reply by nikolausliu** [reply]
tks

**Reply by raineorshine** [reply]
`map` does use more memory since it builds up the return array during execution. In this case it will be `[undefined, undefined, undefined, ...]` so basically negligible. But beware this pattern when the handler returns a value.

You lose two bytes from the added curly braces, but that still leaves you 2 ahead of forEach :).

**Reply by dimaMachina** [reply]
I would suggest replacing `.slice(...).map(...)` with `for..of` because https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-for-each.md
